packages/web/src/pages/oauth-login-page/OAuthLoginPage.tsx

```diff
@@ -83,6 +142,112 @@ const CTAButton = ({
   )
 }
 
+const PermissionsSection = ({
+  scope,
+  isLoggedIn,
+  userEmail
+}: {
+  scope: string | string[] | null
+  isLoggedIn: boolean
+  userEmail?: string | null
+}) => {
+  return (
+    <>
+      <div className={styles.permsTitleContainer}>
+        <h3 className={styles.infoSectionTitle}>
+          {messages.permissionsRequestedHeader}
+        </h3>
+      </div>
+      <div className={styles.tile}>
+        <div className={styles.permissionContainer}>
+          <div
+            className={cn({
+              [styles.visibilityIconWrapper]: scope === 'read'
+            })}
+          >
+            {scope === 'write' ? (
+              <IconPencil
+                className={cn(styles.permissionIcon)}
+                width={18}
+                height={18}
+              />
+            ) : (
+              <IconVisibilityPublic
+                className={cn(styles.permissionIcon, styles.visibilityIcon)}
+                width={21}
+                height={22}
+              />
+            )}
+          </div>
+
+          <div className={styles.permissionTextContainer}>
+            <span className={styles.permissionText}>
+              {scope === 'write'
+                ? messages.writeAccountAccess
+                : messages.readOnlyAccountAccess}
+            </span>
+            {scope === 'read' ? null : (
+              <div className={cn(styles.permissionDetailTextContainer)}>
+                <p
+                  className={cn(
+                    styles.permissionText,
+                    styles.permissionDetailText
+                  )}
+                >
+                  {messages.doesNotGrantAccessTo}
+                  <br />
+                  {messages.walletsOrDMs}
+                </p>
+              </div>
+            )}
+          </div>
+        </div>
+        <div
+          className={cn(
+            styles.permissionContainer,
```

Comment:
fine like this, but you can also use the `composes: permissionContainer` in the style block for `nonFirstPermissionContainer` in the css modules file. lmk if interested in that pattern.